### PR TITLE
[autoscaler] Converting assert False into useful exceptions.

### PR DESCRIPTION
**Pull request by xcharleslin**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/_private/updater.py [hunk 1]
```diff
@@ -235,8 +235,12 @@ def wait_ready(self, deadline):
 
                 cli_logger.print("Running `{}` as a test.", cf.bold("uptime"))
                 first_conn_refused_time = None
-                while time.time() < deadline and \
-                        not self.provider.is_terminated(self.node_id):
+                while True:
+                    if time.time() < deadline:
```
Comment (AmeerHajAli) [inline python/ray/autoscaler/_private/updater.py:239]:
```suggestion
                    if time.time() > deadline:
```

**Review by AmeerHajAli — CHANGES_REQUESTED** [review]
Do you mind adding two pytest checks for this?
Is there a way to avoid "while True"?

File: python/ray/autoscaler/_private/updater.py [hunk 2]
```diff
@@ -235,8 +235,12 @@ def wait_ready(self, deadline):
 
                 cli_logger.print("Running `{}` as a test.", cf.bold("uptime"))
                 first_conn_refused_time = None
-                while time.time() < deadline and \
-                        not self.provider.is_terminated(self.node_id):
+                while True:
+                    if time.time() > deadline:
+                        raise Exception("wait_ready timeout exceeded.")
+                    if self.provider.is_terminated(self.node_id):
```
Comment (AmeerHajAli) [inline python/ray/autoscaler/_private/updater.py:241]: immediately after create_node, is `is_terminated` set to false?

**Review by ericl — APPROVED** [review]
lgtm

**Comment by ericl** [comment]
LINT is failing

**Comment by AmeerHajAli** [comment]
> LINT is failing

I think we should add two pytest checks for this. What do you think?

**Comment by AmeerHajAli** [comment]
> This should be covered by existing tests right? While adding tests is always nice, this PR is just clarifying an exception message and we shouldn't block it by overly expanding the scope to be adding more tests as well.

Can you please help me find where is this tested?
Expanding the scope? This is just 1 unit test with ~10 lines of code.

**Review by AmeerHajAli — APPROVED** [review]
It seems like this is tested here:
https://github.com/ray-project/ray/blob/153dcd3734719d68b0ab2c9c0d4bec1fa85eaf87/python/ray/tests/test_autoscaler.py#L479

**Comment by xcharleslin** [comment]
@ericl @AmeerHajAli Heads up that I don't have merge permissions (also test results are unclear to me - first time in this repo - although I did run the linter).

**Comment by ericl** [comment]
~/build/ray-project/tmp.8pIHIiMtRc
Thu Mar 11 04:23:30 UTC 2021 Flake8....
python/ray/autoscaler/_private/updater.py:244:80: E501 line too long (86 > 79 characters)

(you need to run scripts/format.sh)

**Comment by AmeerHajAli** [comment]
Lint still fails @xcharleslin

**Comment by AmeerHajAli** [comment]
@xcharleslin , can you please rebase + lint this?

**Comment by xcharleslin** [comment]
I am no longer able to push to this PR's branch (due to new repo policies it seems). New PR here:
https://github.com/ray-project/ray/pull/15306

**Pull request closed without merging**